Core/src/GapicRequestWrapper.php

```diff
@@ -0,0 +1,248 @@
+        }
+
+        if ($response instanceof ServerStream) {
+            return $this->handleStream($response);
+        }
+
+        return null;
+    }
+
+    /**
+     * Handles a streaming response.
+     *
+     * @param ServerStream $response
+     * @return \Generator|array|null
+     * @throws Exception\ServiceException
+     */
+    private function handleStream($response)
+    {
+        try {
+            foreach ($response->readAll() as $count => $result) {
+                $res = $this->serializer->encodeMessage($result);
+                yield $res;
+            }
+        } catch (\Exception $ex) {
+            throw $this->convertToGoogleException($ex);
+        }
+    }
+
+    /**
+     * Convert a ApiCore exception to a Google Exception.
+     *
+     * @param \Exception $ex
+     * @return Exception\ServiceException
+     */
+    private function convertToGoogleException($ex)
+    {
+        switch ($ex->getCode()) {
+            case Code::INVALID_ARGUMENT:
+                $exception = Exception\BadRequestException::class;
+                break;
+
+            case Code::NOT_FOUND:
+            case Code::UNIMPLEMENTED:
+                $exception = Exception\NotFoundException::class;
+                break;
+
+            case Code::ALREADY_EXISTS:
+                $exception = Exception\ConflictException::class;
+                break;
+
+            case Code::FAILED_PRECONDITION:
+                $exception = Exception\FailedPreconditionException::class;
+                break;
+
+            case Code::UNKNOWN:
+                $exception = Exception\ServerException::class;
+                break;
+
+            case Code::INTERNAL:
+                $exception = Exception\ServerException::class;
+                break;
+
+            case Code::ABORTED:
+                $exception = Exception\AbortedException::class;
+                break;
+
+            case Code::DEADLINE_EXCEEDED:
+                $exception = Exception\DeadlineExceededException::class;
+                break;
+
+            default:
+                $exception = Exception\ServiceException::class;
+                break;
+        }
+
+        $metadata = [];
+        if ($ex->getMetadata()) {
+            foreach ($ex->getMetadata() as $type => $binaryValue) {
+                if (!isset($this->metadataTypes[$type])) {
+                    continue;
+                }
+                $metadataElement = new $this->metadataTypes[$type];
+                $metadataElement->mergeFromString($binaryValue[0]);
+                $metadata[] = $this->serializer->encodeMessage($metadataElement);
+            }
+        }
+
+        return new $exception($ex->getMessage(), $ex->getCode(), $ex, $metadata);
+    }
+
+    /**
+     * Helper function that transforms some $config params to another.
+     * 
+     * @param array $config Input config params to be transformed.
+     * @return array
+     */
+    private function convertConfigParams(array $config)
```

Comment:
This was written earlier when the intention was to support `restOptions` and `grpcOptions`.
But since we don't need to do that as discussed in the design doc, we can remove this method call completely.